project.yaml

```diff
@@ -907,14 +907,47 @@ Layer:
               ('railway_' || (CASE WHEN railway IN ('platform') THEN railway ELSE NULL END)),
               (('aeroway_' || CASE WHEN aeroway IN ('runway', 'taxiway', 'helipad') THEN aeroway ELSE NULL END))
             ) AS feature
-          FROM planet_osm_polygon
-          WHERE highway IN ('residential', 'unclassified', 'pedestrian', 'service', 'footway', 'living_street', 'track', 'path', 'platform', 'services')
-            OR railway IN ('platform')
-            OR aeroway IN ('runway', 'taxiway', 'helipad')
+          FROM (
+          SELECT 
+              way, highway, railway, aeroway, z_order, way_area FROM planet_osm_polygon
+            WHERE highway IN ('residential', 'unclassified', 'pedestrian', 'service', 'footway', 'living_street', 'track', 'path', 'platform', 'services')
+              OR railway IN ('platform')
+              OR aeroway IN ('taxiway', 'helipad')
+          UNION ALL
+          SELECT -- runways with width either tagged or estimated from runway length
+              ST_Buffer(
```

Comment:
Nice.  I had considered doing ST_Buffer() on geography originally but dismissed that due to the internal complexity and the lack of support for endcap/join.  But this way it looks like it could work.  Nice catch with the need for an extended bounding box as well - this would apply regardless of the method to buffer.

I will try to integrate that with the rest later.  I still think from the purely mathematical standpoint my original approach is the most straightforward.  The technique above involves quite a lot of ultimately unnecessary coordinate transforms in the standard situation with mercator coordinates but you are probably right it is the only way to equally support geographic and projected coordinates.  It can also probably be considered the intended method from the postgis side and as they say: When in Rome do like the Romans do...